core/kernel/classes/class.Resource.php

```diff
@@ -187,40 +176,32 @@ public function getTypes()
      * method preventing to call the get getPropertyValues() method for a such common
      * operation.
      *
-     * @author Patrick Plichart <patrick@taotesting.com>
      * @return string A Uniform Resource Identifier (URI).
+     * @throws core_kernel_persistence_Exception
+     * @author Patrick Plichart <patrick@taotesting.com>
      */
-    public function getLabel()
-    {
-        if (is_null($this->label)) {
-            $label =  $this->getOnePropertyValue($this->getProperty(OntologyRdfs::RDFS_LABEL));
-            $this->label = is_null($label)
-                ? ''
-                : ($label instanceof core_kernel_classes_Resource
-                    ? $label->getUri()
-                    : (string)$label
-                   )
-            ;
+    public function getLabel(): ?string
```

Comment:
I don't see a way of getting a `null` as a return value here.

```suggestion
    public function getLabel(): string
```